nncf/torch/engine.py

```diff
@@ -32,12 +32,17 @@ def __init__(self, model: nn.Module):
         self._model = model
         self._model.eval()
 
-    def infer(self, input_data: Union[torch.Tensor, Dict[str, torch.Tensor]]) -> Union[torch.Tensor, Dict[str, Any]]:
+    def infer(
+        self, input_data: Union[torch.Tensor, Tuple[torch.Tensor], Dict[str, torch.Tensor]]
+    ) -> Union[torch.Tensor, Dict[str, Any]]:
         """
         Runs Torch model on the provided input.
 
         :param input_data: inputs for the model
         :return output_data: model outputs
         """
-
+        if isinstance(input_data, dict):
+            return self._model(**input_data)
+        if isinstance(input_data, tuple):
+            return self._model(*input_data)
```

Comment:
Which tests caught this?

Reply:
'tests/torch/test_transform_fn.py'
But there was missed `dict` as input, added it.

Reply:
Looks like this changes conflicts with https://github.com/openvinotoolkit/nncf/pull/2196, dict and tuple will supported after your PR?

Reply:
I guess we'll have to see, based on your tests.